lib/src/validator/relative_version_numbering.dart

```diff
@@ -47,15 +49,14 @@ class RelativeVersionNumberingValidator extends Validator {
     final previousOptedIn = _optedIntoNullSafety(previousPubspec);
 
     if (currentOptedIn && !previousOptedIn) {
-      warnings.add(
-          'You are about to publish a package opting into null-safety.\n'
+      hints.add('You are about to publish a package opting into null-safety.\n'
           'The latest version ${previousVersion.version} has not opted in.\n'
           'Be sure to read $guideUrl for best practices.');
```

Comment:
```suggestion
      hints.add('You\'re about to publish a package that opts into null safety.\n'
          'The previous version (${previousVersion.version}) wasn\'t opted in.\n'
          'See $guideUrl for best practices.');
```

Or maybe _previous ... wasn't_ -> _current ... isn't_?

Or maybe something like this:

The package you're about to publish is opting into null safety for the first time.
Before you publish, make sure you've followed the best practices at $guideUrl.

(What exactly do we want to make sure they've done correctly before publishing their newly null-safe package?)

Reply:
> (What exactly do we want to make sure they've done correctly before publishing their newly null-safe package?)

One thing that comes to mind is explaining about compatibility mode, and why you would want to avoid it.

I'm not sure we know everything yet. From the associated issue:

> When a null-safety compliant version of an existing package is published for the first time, we should provide a warning with a link to the null-safety migration guide.
>
> There are many pitfalls one can fall into when migrating, ideally the migration guide would have a check list and we can reference this check list.
>
> note. We want to surface a warning containing a URL to the migration guide, this way the migration guide can be adjusted as lessons are learned.